### PR TITLE
WAZO-1081: Google more contact info

### PR DESCRIPTION
**Pull request by bloom1**
Extract and expose more info from the Google source:

- [x] First Name
- [x] Name
- [x] Enterprise
- [x] Function
- [x] All tel and label
- [x] All Email and label
- [x] All addresses
- [x] Note

### CODE REVIEW

File: integration_tests/suite/test_google_contacts.py [hunk 1]
```diff
@@ -160,17 +183,25 @@ def test_list(self, google_api):
...
                         emails=contains_inanyorder('mario@bros.example.com'),
+                        email_labels=contains_inanyorder('other'),
                         numbers=contains_inanyorder('+15555551111', '+15555551234'),
                         numbers_by_label=has_entries(
                             home='+15555551111', mobile='+15555551234'
                         ),
+                        organizations=contains_inanyorder('MarioLand'),
+                        organization_titles=contains_inanyorder('Artist'),
+                        addresses=contains_inanyorder('Main Land', 'Alternative Land'),
+                        address_labels=contains_inanyorder('home', 'Second address'),
```
Comment (fblackburn1) [inline integration_tests/suite/test_google_contacts.py]: After IRL discussion, labels should be merged with objects inside a list of dictionnary, not a list of string

**Comment by bloom1** [comment]
recheck

**Comment by fblackburn1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
@fblackburn1 it failed because of the display that uses the old email format - would it be better to leave the old emails field and add emails_with_labels or something like that? Or to update the display?

File: wazo_dird/plugins/google_backend/plugin.py [hunk 2]
```diff
@@ -121,9 +121,15 @@ def _first_match_predicate(self, term, contact):
...
-                        return True
+                    if isinstance(value, dict):
+                        for sub_value in value.values():
+                            if term == sub_value.lower():
+                                return True
+                    else:
+                        if term == value.lower():
+                            return True
             else:
+                logger.debug('Item is not a dict or list')
```
Comment (fblackburn1) [inline wazo_dird/plugins/google_backend/plugin.py]: We can remove it.
But just wait until zuul complete its check and we will force push/merge if need to be in this release

Comment (bloom1) [reply wazo_dird/plugins/google_backend/plugin.py]: Oops, did not mean to leave it in. I will force push without it.

**Pull request merged**